cmd/project_browse.go

```diff
@@ -22,13 +44,15 @@ var projectBrowseCmd = &cobra.Command{
 			log.Fatal(err)
 		}
 
-		err = browse(p.WebURL)
+		err = browse(projectBrowseGetPath(p.WebURL, p.DefaultBranch, projectBranch, projectFile))
 		if err != nil {
 			log.Fatal(err)
 		}
 	},
 }
 
 func init() {
+	projectBrowseCmd.Flags().StringVar(&projectFile, "file", "", "path to specified file")
+	projectBrowseCmd.Flags().StringVar(&projectBranch, "branch", "", "specific branch (overrides default branch)")
```

Comment:
Rather than a branch, could this take a git ref such that it would work if the user supplied a git tag or a git SHA for example?

Also, when omitted to, could this default to `git rev-parse HEAD` of the git repo that the calling process is currently located?

Reply:
>Rather than a branch, could this take a git ref such that it would work if the user supplied a git tag or a git SHA for example? 

Sure, can you provide me an example URL so I can reference it?

> Also, when omitted to, could this default to git rev-parse HEAD of the git repo that the calling process is currently located?

This gets very tricky when dealing with local git repositories.  'git rev-parse HEAD' may or may not point to a valid remote.

Reply:
`https://${HOSTNAME}/${PATH}/-/tree/${REF}` should work. `${REF}` can be a git SHA, a tag, or a branch.

That's a good point about `git rev-parse HEAD` not being present on the remote since the local update might not have been pushed yet.

Reply:
Let me add a --ref option.  I think that is different from a branch so it needs its own option.

Reply:
I tested `https://${HOSTNAME}/${PATH}/-/tree/${REF}` where the `${REF}` was a branch name. I think we should only need the `--ref` option and no longer need the `--branch` option.

Reply:
I think we need both -- users could request to see a file on a pushed branch :)

Reply:
Wouldn't `--ref` also work for seeing a file on a pushed branch?

Reply:
Here are some examples showing branch, tag, and SHA:

|  type | example |
|---|---|
| branch  |   https://gitlab.com/gitlab-org/gitlab-foss/-/blob/doc/test-naming/.gitlab-ci.yml |
| tag | https://gitlab.com/gitlab-org/gitlab-foss/-/blob/v15.11.11/.gitlab-ci.yml |
| SHA | https://gitlab.com/gitlab-org/gitlab-foss/-/blob/55a6d7c9/.gitlab-ci.yml |